### ISSUE TITLE
Feature ReQuestion: References / links support?

### ISSUE THREAD

**Issue by flyinbutrs** [keepassxreboot/keepassxc#433]
I'm not sure if this is already supported or if this is a feature request, but is there a way to create / utilize references like [described here](https://superuser.com/a/870509/74640) for Windows keepass2?

**Comment by TheZ3ro** [comment]
Currently there isn't a nice GUI for inserting references beside when duplicating/cloning an entry. (Like https://superuser.com/a/434046) But if you manually insert the reference code and KeePassXC will interpret the reference correctly

**Comment by flyinbutrs** [comment]
Could you elaborate on how to do that exactly? What would I put as the password in entry A to reference the password for entry B?

**Comment by TheZ3ro** [comment]
@flyinbutrs Yes, the complete reference format is described on [KeePass website](http://keepass.info/help/base/fieldrefs.html). 

Currently you will need to build KeePassXC form source from the develop branch and the only way to link references is by using entry UUID.

So if you want to do this right now you have to:
- Go in the entry you want to reference and from the *Property* section copy the *UUID*
- Create a new entry 
- Set the attribute you want to reference to `{REF:<WantedField>@I:<UUID>}` and pasting the UUID.

`WantedField` can only be a standard field:
```
T	Title
U	User name
P	Password
A	URL
N	Notes
```

Later I will add this to the wiki maybe

**Comment by flyinbutrs** [comment]
Not discounting user error, but I don't think this is working. I put this into a password field (twice):

{REF:P@I:77d14b5fb035281bc4d2781e006cf386}

And have double checked that the UUID is correctly pointing to another entry, but when I go to the page, both keepasshttp, autotype, and command+c put that exact string into the password field / clipboard rather than the referenced password.

I'm using keepassxc 2.1.3 on OSX, if it matters.

**Comment by phoerious** [comment]
2.1.3 doesn't support references yet. You need to compile the develop branch for that.

**Comment by flyinbutrs** [comment]
Oops... reading comprehension failure. Is that coming in the next release? I'd prefer not to compile from source if it's only going to be a few weeks.

**Comment by phoerious** [comment]
We don't have a release date yet and some major features are yet to be completed.

**Comment by Felixoid** [comment]
Hi, any progress here?

What about to add submenu `Copy reference to clipboard` here?
![screen_2018-04-12_13-47-27](https://user-images.githubusercontent.com/3025537/38675934-a7ae811a-3e59-11e8-8a99-e33f139b1815.png)

**Comment by lagagain** [comment]
@Felixoid 
I like the function in keepass2, and add to submenu is a good idea, but in my use cases, frequently, are that need add a account which use OAuth login(like login with google, facebook or github).  Because that, I don't think the way(just add to submenu) is very sense to me.

**Comment by zioalex** [comment]
Hi Guys, any update here. This feature is very convinient and I miss it coming from KeePass.

**Comment by ftorto** [comment]
Isn't it a duplicate of #2030 ?

**Comment by Felixoid** [comment]
Looks like otherwise, @ftorto

**Comment by mcflypl** [comment]
I use AD login on many pages and I miss this feature...

**Comment by stefan123t** [comment]
@flyinbutrs I like your idea :+1: of having several (if not all) Field Reference Entries available to insert at the current cursor position / at the end of  the Use custom Auto-Type sequence: Input field. ~~while it could even be extended to have some {COMMAND} completion as suggested in #6782 and #6462~~ :innocent:

Configuring Auto-Type Sequence in the dialog usually requires reading the documentation sections under
[Configure Auto-Type Sequences](https://keepassxc.org/docs/KeePassXC_UserGuide.html#_configure_auto_type_sequences) and
[Searching the Database](https://keepassxc.org/docs/KeePassXC_UserGuide.html#_searching_the_database)

The dialog still has plenty of space under the `[ ] use a specific sequence for this association` input box to at least add those two documentation links and place some insert buttons for the feature requested here:
![grafik](https://user-images.githubusercontent.com/11159753/135656198-bfaed4ab-632c-47c7-aa55-c889bc591190.png)

The most frequently used / known ones are IMHO as they are also supported in KeePass, KeePassX and KeePassXC:
[Field Reference](https://keepass.info/help/base/fieldrefs.html) / [Placeholders](https://keepass.info/help/base/placeholders.html) | Description
-- | --
`{USERNAME}{TAB}{PASSWORD}{ENTER}` | Default (Username & Password)
`{USERNAME}` | Username
`{TAB}` | Tab
`{PASSWORD}` | Password
`{ENTER}` | Enter
`{TOTP}` | Current TOTP
`{DELAY 2500}` | Delay of 2500 ms
`{CLEARFIELD}` | Clear the current Input field
`{S:<ATTR_NAME>}` | Additional Attribute ATTR_NAME
`{REF:U@I:<UUID>}` | Username of another Entry UUID
`{REF:P@I:<UUID>}` | Password of another Entry UUID
`{REF:<FIELD>@<SEARCH_IN>:<SEARCH_TEXT>}` | FIELD from another Entry with Field SEARCH_IN matching SEARCH_TEXT

* `ATTR_NAME` should be a pull-down offering all the configured Additional Attributes of the Entry, regardless whether Protected or not.

* `FIELD` and `SEARCH_IN` should be pull down boxes offering the following Options:

[Fields](https://keepassxc.org/docs/KeePassXC_UserGuide.html#_modifiers_and_fields) | Name
-- | --
`T` | Title
`U` | User name
`P` | Password
`A` | URL
`N` | Notes
`I` | UUID
`O` | Other custom strings (KeePass 2.x only)

* `UUID` or `SEARCH_TEXT` should allow to using an `...` Ellipsis Button to select / search one of the entries from the Database in a separate dialog and select its Name, UUID, etc. as value.
There could be further hints to the Regex Options in the UUID / SEARCH_TEXT Field as documented under 



[Modifiers](https://keepassxc.org/docs/KeePassXC_UserGuide.html#_modifiers_and_fields) | Description
-- | --
`-` | Exclude this term from results
`!` | Exclude this term from results
`+` | Match this term exactly
`*` | Term is handled as a regular expression

[Wild Card Character](https://keepassxc.org/docs/KeePassXC_UserGuide.html#_wild_card_characters_and_logical_operators) | Description
-- | --
`*` | Match anything
`?` | Match one character
`\|` | Logical OR

[Sample Search Query](https://keepassxc.org/docs/KeePassXC_UserGuide.html#_sample_search_queries) | Description
-- | --
`user:johnsmith url:www.google.com` | Searches the Username field for johnsmith and the URL field for www.google.com.
`user:john\|smith` | Searches the Username field for john OR smith.
`+user:johnsmith -url:www.google.com *notes:"secret note \d"` | Search  the username field for exactly johnsmith, the URL must not contain  www.google.com, and notes contains secret note [digit].

**Comment by yaoyingkeji** [comment]
Export file (xml、html、csv), reference location, not reference content, but code?
![截图_选择区域_20231029190546](https://github.com/keepassxreboot/keepassxc/assets/93997433/7b34b6f3-156d-484e-acd0-fd2d0332d1ef)

**Comment by yaoyingkeji** [comment]
Using "{REF:T@I:c634a207451b4643ac8daf8ef44f40cb}" in Additional attributes has no effect.
![图片](https://github.com/keepassxreboot/keepassxc/assets/93997433/20eabffa-d7f3-4650-ab06-00e4b9365bf1)

**Comment by Silverlan** [comment]
Would also like to see this implemented!

**Comment by aetonsi** [comment]
thank you @droidmonkey for adding this to the WIP Tracker!

**Comment by stefan123t** [comment]
Regarding display of `{REF:<FIELD>@I:<UUID>}` I would kindly ask to consider some information on what this UUID is actually pointing to, e.g. adding a tooltip or comment `# field from group > entry name (uuid)`.
I always wonder which entry this UUID is actually referring to when viewing / editing existing reference entries.

Regarding export (and editing ?) @yaoyingkeji may have a valid point in that it should be possible to resolve REF entries and replace them with the values from the reference instead. As not all export formats or purposes may still be able to resolve the REF uuids afterwards.
